LoRaEngine/LoraKeysManagerFacade/FunctionsBundler/DeduplicationExecutionItem.cs

```diff
@@ -69,9 +81,48 @@ internal async Task<DuplicateMsgResult> GetDuplicateMessageResultAsync(DevEui de
 
                         if (updateCacheState)
                         {
+                            var previousGateway = cachedDeviceState.GatewayId;
+
                             cachedDeviceState.FCntUp = clientFCntUp;
                             cachedDeviceState.GatewayId = gatewayId;
                             _ = deviceCache.StoreInfo(cachedDeviceState);
+
+                            logger?.LogDebug("Previous connection owner was '{PreviousConnectionOwner}', current message was received from '{Gateway}'", previousGateway, gatewayId);
+                            if (previousGateway != gatewayId)
+                            {
+                                var loraC2DMessage = new LoRaCloudToDeviceMessage()
+                                {
+                                    DevEUI = devEUI,
+                                    Fport = FramePort.AppMin,
+                                    MessageId = Guid.NewGuid().ToString()
+                                };
+
+                                using var scope = logger?.BeginScope(new Dictionary<string, object> { [MessageIdKey] = loraC2DMessage.MessageId });
```

Comment:
why do we create a scope here? The scope makes sense for me on a higher level in the stack to track common properties, like the DevEui or the calling LNS id.

Reply:
every log statement contains the DevEui. @bastbu was looking into adding logging scopes to the function API. Is this overlapping?

Reply:
I created the scope to track all logs associated with that message id but since we will remove all logs but 1, I will remove the scope as well. Btw I don't think every log statement in the function contains the DevEui (at least I don't see it)

Reply:
d489a749d8a517f62eb764a7522091ec36527b1c